packages/end-to-end-testing/lib/chains.ts

```diff
@@ -0,0 +1,243 @@
+import { getAddress, isNetworkName, ZetaNetworkName } from "@zetachain/addresses";
+import { isLocalNetworkName, NetworkName } from "@zetachain/addresses/src/addresses.helpers";
+import { getConnectorFactory, getTokenFactory } from "@zetachain/protocol-contracts/lib/contracts.helpers";
+import {
+  ZetaConnectorEth,
+  ZetaConnectorNonEth,
+  ZetaEth,
+  ZetaNonEth,
+} from "@zetachain/protocol-contracts/typechain-types";
+import { Axios, AxiosInstance } from "axios";
+import * as dotenv from "dotenv";
+import { Overrides, providers, Signer, Wallet } from "ethers";
+import { ethers } from "hardhat";
+const axios = require("axios").default;
+// dotenv.config();
+
+// console.debug = function () {}; // Disables Debug Level Logging
+// console.info = function () {}; // Disables Info Level Logging
+
+const defaultOverrideOptions: Overrides = {
+  gasLimit: 2500000,
+  gasPrice: 20000000000,
+};
+
+export class Blockchain {
+  name: NetworkName | ZetaNetworkName;
+  chainId: number;
+  rpcEndpoint: string;
+  initStatus: undefined | Promise<void>;
+  p: any;
```

Comment:
Can we call this provider? We can also type it by using something like `typeof ethers.providers.JsonRpcProvider`